lib/faraday/connection.rb

```diff
@@ -464,5 +442,16 @@ def set_authorization_header(header_type, *args)
         header(*args)
       headers[Faraday::Request::Authorization::KEY] = header
     end
+
+    def find_default_proxy
+      warn 'no_proxy is unsupported' if ENV['no_proxy'] || ENV['NO_PROXY']
+      uri = ENV['http_proxy']
+      if uri && !uri.empty?
+        uri = 'http://' + uri if uri !~ /^http/i
+        return uri
+      else
+        return nil
+      end
```

Comment:
I agree. Moreover, the `else` branch is completely redundant. so the `if` can be simplified into:

```ruby
if uri && !uri.empty?
  uri = 'http://' + uri if uri !~ /^http/i
  uri
end
```